scripts/null_route_helper

```diff
@@ -0,0 +1,216 @@
+#!/usr/bin/env python3
+
+"""
+Utility for blocking and unblocking traffic from given source ip_prefix on ACL tables.
+
+The block operation will insert a DENY rule at the top of the table. The unblock operation
+will remove an existing DENY rule that has been created by the block operation (i.e. it does
+NOT insert an ALLOW rule, only removes DENY rules).
+
+Since SONiC supports multi ACL rules share the same priority, all ACL rules created by null_route_helper will
+use the highest priority(9999).
+
+Example:
+
+Block traffic from 10.2.3.4/32:
+./null_route_helper acl_table_name block 10.2.3.4/32
+
+Unblock all traffic from 10.2.3.4/32:
+./null_route_helper acl_table_name unblock 10.2.3.4/32
+"""
+
+
+from __future__ import print_function
+
+import syslog
+import sys
+import click
+import ipaddress
+
+from swsssdk import ConfigDBConnector
+
+
+CONFIG_DB_ACL_TABLE_TABLE = "ACL_TABLE"
+CONFIG_DB_ACL_RULE_TABLE = "ACL_RULE"
+CONFIG_DB_VLAN_TABLE = "VLAN"
+
+ACTION_ALLOW = "FORWARD"
+ACTION_DENY = "DROP"
+
+# Since SONiC supports multi ACL rules share the same priority, we use 9999 (the highest) for all rules
+ACL_RULE_PRIORITY = 9999
+# The key of rule will be overridden with BLOCK_RULE_ + ip
+ACL_RULE_PREFIX = 'BLOCK_RULE_'
+
+def notice(msg):
+    """
+    Log a NOTICE message to the console and syslog
+    """
+    syslog.syslog(syslog.LOG_NOTICE, msg)
+    print(msg)
+
+
+def error(msg):
+    """
+    Log an ERR message to the console and syslog, and exit the program with an error code
+    """
+    syslog.syslog(syslog.LOG_ERR, msg)
+    print(msg, file=sys.stderr)
+    sys.exit(1)
+
+
+def ip_ver(ip_prefix):
+    return ipaddress.ip_network(ip_prefix, False).version
+
+
+def check_table_existence(configdb, table_name):
+    """
+    Check the existence of pre-created ACL tables
+    """
+    target_table = configdb.get_entry(CONFIG_DB_ACL_TABLE_TABLE, table_name)
+
+    if not target_table:
+        error("Table {} not found, exiting...".format(table_name))
+
+    return target_table
+
+
+def get_acl_rule_key(ip_prefix):
+    """
+    Get the key that will be used to refer to the ACL rule used to block traffic from a source ip.
+    Since the rules are all given the same priority in SONiC, we can't identify a rule based on the priority.
+    So, we use the destination IP being blocked to give each rule a unique name in the system.
+    """
+    return ACL_RULE_PREFIX + str(ip_prefix)
+
+
+def get_all_acl_rules(configdb, table_name):
+    """
+    Return a dict of existed acl rules
+    {(u'SVI_PROTECT_V4', u'BLOCK_RULE_0.0.0.0/0'): {'PRIORITY': '9999', 'PACKET_ACTION': 'FORWARD', 'SRC_IP': '0.0.0.0/0', 'DST_IP': '0.0.0.0/0'},...}
+    """
+    key = CONFIG_DB_ACL_RULE_TABLE + '|' + table_name
+    all_rules = configdb.get_table(key)
+    block_rules = {}
+    for k, v in all_rules.items():
+        if k[1].startswith(ACL_RULE_PREFIX):
+            block_rules[k] = v
+
+    return block_rules
+
+
+def validate_input(ip_address):
```

Comment:
Can you reuse anything from utils?

Reply:
The ```validate_input``` in this PR will add prefix len for IP addresses that don't explicitly set prefix. For example, ```1.2.3.4``` is changed to ```1.2.3.4/32```. The common utils may not do this.